vmangos/core
# Sunken Temple: Rework Atal'ai Statue Event

## 🍰 Pullrequest
- Prevent quest turn in of [Secret of the Circle](https://www.wowhead.com/classic/quest=3447/secret-of-the-circle) while boss Atal'alarion is alive.
- fix object state of Idol of Hakkar after server crash/restart
- fix state of Atal'ai Statue objects after crash/restart
- fix Atal'alarion visibility when alive after server crash/restart
- fix Atal'ai Statue traps getting triggered when statue in correct order was used
- some code cleanup

### Proof
https://www.wowhead.com/classic/quest=3447/secret-of-the-circle#comments
_Killing the guardian will allow you to click on the statue in the center of the pool to complete the quest._

### Issues
- None

### How2Test
- quest add 3447
- check circle traps
- activate circle (S, N, SW, SE, NW, NE)
- while Atal'alarion is alive, it should not be possible to interact with the Idol for quest turn in
- stop server and check dungeon state after restart

### Todo / Checklist
